### PR TITLE
Chore/metrics advanced

### PR DESCRIPTION
**Pull request by pitoniak32**
Issue That prompted these changes: https://github.com/open-telemetry/opentelemetry-rust/issues/1060

References:
- mention of wanting alternate Temporality example: https://github.com/open-telemetry/opentelemetry-rust/issues/1060#issuecomment-1834421336
- mention of duplication in benches: https://github.com/open-telemetry/opentelemetry-rust/pull/1057#discussion_r1205798111

## Changes
`DeltaTemporalitySelector`:
- Moving the `DeltaTemporalitySelector` to same file as `DefaultTemporalitySelector`.
- make the visibility of the struct `pub` so its usable for pipeline setup.

Example:
- Update the advanced metrics example to use `DeltaTemporalitySelector`.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-otlp/src/metric.rs [hunk 1]
```diff
@@ -169,7 +169,7 @@ where
     ///
     /// [exporter-docs]: https://github.com/open-telemetry/opentelemetry-specification/blob/a1c13d59bb7d0fb086df2b3e1eaec9df9efef6cc/specification/metrics/sdk_exporters/otlp.md#additional-configuration
     pub fn with_delta_temporality(self) -> Self {
-        self.with_temporality_selector(DeltaTemporalitySelector)
+        self.with_temporality_selector(DeltaTemporalitySelector::new())
```
Comment (cijothomas) [inline opentelemetry-otlp/src/metric.rs:172]: the future of these methods in OTLP crate is unknown based on https://github.com/open-telemetry/opentelemetry-rust/pull/1811
However, this change itself is fine.

File: examples/metrics-advanced/src/main.rs [hunk 2]
```diff
@@ -44,7 +45,11 @@ fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
         }
     };
 
-    let exporter = opentelemetry_stdout::MetricsExporterBuilder::default().build();
+    // Build exporter using Delta Temporality.
```
Comment (cijothomas) [inline examples/metrics-advanced/src/main.rs:48]: While the concept is somewhat advanced, it is a common ask. Could you make this in the metrics-basic itself, as a code comment, so users can pick between delta vs cumulative by removing commented out code?

(It is possible that we'll have different ways to configure this in the future, but this change is still very good)

**Review by cijothomas — APPROVED** [review]
LGTM.
If you can make the change in metrics-basic, that'd be sweet. It is okay to do it in a follow up as well.

**Comment by pitoniak32** [comment]
> LGTM.
> 
> If you can make the change in metrics-basic, that'd be sweet. It is okay to do it in a follow up as well.

Awesome! I will make the requested changes for this tomorrow for the example.

**Pull request merged**